### PR TITLE
Receive and save the latest media files.

### PR DESCRIPTION
**Pull request by bondrogeen**

### CODE REVIEW

**Comment by Apollon77** [comment]
I had a look into it. I personally would prefer that the directory for storage can be configured via an adapter config option like for other adapters. So allow absolute path's and all relative path should be automatically relative to "iobroker-data" directory. Can you change that please that way? Thank you for your contribution.

You can check iobroker.history for the "path-logic" and how to get the path

**Comment by bondrogeen** [comment]
Okay, I'll add the ability to change the path to the folder.
Is the default path changed from "opt / iobroker / tmp" to "opt / iobroker / iobroker-data / tmp"?

**Comment by Apollon77** [comment]
Why temp?! use pushover-media or such? :-)

**Comment by bondrogeen** [comment]
I planned to store only the last received files from telegrams.
For example, the ability to send voice to recognition.

Do you want to use adapter.telegram to send and store media files?

**Comment by Apollon77** [comment]
I have no need at all for medias (currently), only thoughtabout possible usage.
When you only store one file ... who cleans the directory?

**Comment by bondrogeen** [comment]
The last files are overwritten each time.
Just stored in the directories tmp/nameadapter_ [index]

File: main.js [hunk 1]
```diff
@@ -423,6 +437,104 @@ function sendMessage(text, user, chatId, options) {
     return count;
 }
 
+function saveFile(file_id, fileName, callback) {
+    bot.getFileLink(file_id).then(function (url) {
+        adapter.log.debug('Received message: ' + url);
+        https.get(url, function (res) {
+            if (res.statusCode == 200) {
```
Comment (GermanBluefox) [inline main.js]: if (res.statusCode === 200)

File: io-package.json [hunk 2]
```diff
@@ -244,6 +244,18 @@
                 "write": false
             },
             "native": {}
+        },
+        {
+            "_id": "communicate.pathFile",
+            "type": "state",
+            "common": {
+                "role": "text",
+                "name": "Users as json, which are constantly authenticated",
```
Comment (GermanBluefox) [inline io-package.json]: ??? Is the name right?

File: main.js (hunk 1 above)
Comment (bondrogeen) [reply main.js]: corrected the mistake

File: io-package.json (hunk 2 above)
Comment (bondrogeen) [reply io-package.json]: corrected the mistake

**Pull request merged**